### DISCUSSION TITLE
开源世界当中到底存不存在“白嫖”？ | 夜天之书

### DISCUSSION THREAD

**Discussion by giscus** [tisonkun/blog#30] [Comments]
# 开源世界当中到底存不存在“白嫖”？ | 夜天之书

开源软件不是凭空出现的，开发开源软件是一项艰苦卓绝的工作。每个开源软件的背后少则有原作者一人的投入，多则协同了成千上万人组成的开源社群的共同努力。然而，开源软件的源代码总是免费可得，并且开源软件协议总是不限制用户的使用形式。 基于开源软件完成工作乃至搭建业务盈利的用户，并不总是参与软件开发的人，这种形似经济学中“搭便车”的行为，在国内被提及的时候总会被称为“白嫖”，以至于后者称为圈内的一个热词。那

https://www.tisonkun.org/2022/07/05/open-source-free-rider/

**Comment by PuzoLiang** [comment]
Stallman 发起free software一个是对于分享的信仰，还有一个我认为是Stallman is a rich man或者是对物质追求不是那么剧烈的人，他对于自己感兴趣的事情或者观点比拿这个做商业上的考虑要多的多，甚至在某种意义上，他并不care 他的软件是否有人拿去盈利，我记得他曾经还搞过一个找出bug奖励多少美金的趣事儿。 开源软件中还有一个主题是很多时候开源作者的dev水平很高，但是如何转换成生产力甚至客户价值而来的营收却一窍不通，而有人可以做到，这是让我最惊叹的地方。 开源的趋势我认为以后分两派，一派是stallman 为首的，重精神而轻物质，完全的apache2.0协议，爱干嘛干嘛去，至少留下我的名字让我有点荣誉感就OK。一派 就是ELK 为首的 SSPL 协议，或者mongodb之类的 mongodb 协议，这种开源软件背后的主体是商业公司，我承诺我维护开源，并且被大量的to B toC 的用户使用，各方对软件的功能和bug修复需求强烈，这种 runtime changable 的软件就应该对 toB的 收费，至少拿出点真金白银来回馈社区才能对得起开源这么多人的努力吧，第二种衍生协议什么toB的企业要是修改也必须开源这个就弱化许多了，东道主肯定更希望的是to B企业拿出钱来回馈。 总结下来就是，开源社区应该对to B的收费或者授权，至少给了原作者经济上的激励，对于toC 仍旧 免费维护开源的声誉，这样的结果是最合适的，jetbrains did， aws did and microsoft did， awesome，当然，stallman这种free forever的除了仰望 no words to say.

**Reply by xxchan** [reply]
建议阅读[《企业实践开源的动机》](https://www.tisonkun.org/2022/05/26/motivation-of-enterprise-open-source/)，开源商业化并不只有卖授权一种模式，并且作者其实认为直接销售开源软件的商业模式终将被证明是不可行的。
